File: core/src/main/java/com/core/dagger/DaggerBottomSheetFragment.java
package com.core.dagger;

import android.content.Context;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.AndroidSupportInjection;
import dagger.android.support.HasSupportFragmentInjector;

/**
 * @author volkanhotur
 */

public class DaggerBottomSheetFragment extends BottomSheetDialogFragment implements HasSupportFragmentInjector {

    @Inject
    DispatchingAndroidInjector<Fragment> childFragmentInjector;

    @Override
    public void onAttach(@NonNull Context context) {
        AndroidSupportInjection.inject(this);
        super.onAttach(context);
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return childFragmentInjector;
    }
}

File: core/src/main/java/com/core/dagger/DaggerBottomSheetFragment.kt
package com.core.dagger

import android.content.Context
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasAndroidInjector
import dagger.android.support.AndroidSupportInjection
import javax.inject.Inject

/**
 * @author volkanhotur
 */
open class DaggerBottomSheetFragment : BottomSheetDialogFragment(), HasAndroidInjector {

    @JvmField
    @Inject
    var childFragmentInjector: DispatchingAndroidInjector<Any>? = null

    override fun onAttach(context: Context) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun androidInjector(): AndroidInjector<Any> {
        return childFragmentInjector!!
    }
}